lib/manageiq_performance/middleware_storage/file.rb

```diff
@@ -54,6 +54,34 @@ def format_path_for_filename path
       def request_timestamp env
         env['HTTP_MIQ_PERF_TIMESTAMP'] || (Time.now.to_f * 1000000).to_i
       end
+
+      private
+
+      def format_data data
+        case data
+        when Hash
+          if data[:queries]
+            ast = Psych::Visitors::YAMLTree.create
+            ast << data
+            ast.tree.grep(Psych::Nodes::Mapping).each do |mapping|
```

Comment:
This is a bit of a performance it doing this, so it might be something that we want to be an opt in.  On a 12k line yaml file, we are talking about 30-50ms of extra processing.  Not brutal, but this example with 500ish queries made, each with full stacktraces, and our app can make results that are much bigger.

Something I am considering, but might just merge and iterate later.